zstacklib/zstacklib/utils/gpu.py

```
@@ -661,6 +662,37 @@ def has_nvidia_gpu():
     return r == 0 and o and len(o.strip()) > 0
 
 
+@lock.lock("gpu-detach-from-host")
+def detach_from_host_with_lock(addr, vendor):
+    pre_detach_from_host(vendor)
+    r, o, e = bash_roe("virsh nodedev-detach pci_%s" % addr.replace(':', '_').replace('.', '_'))
```

Ruff (0.14.8)

`bash_roe` may be undefined, or defined from star imports

(F405)